src/server/config/transform_deprecations.js

```diff
@@ -25,6 +25,18 @@ const savedObjectsIndexCheckTimeout = (settings, log) => {
   }
 };
 
+const rewriteBasePath = (settings, log) => {
+  if (_.has(settings, 'server.basePath') && !_.has(settings, 'server.rewriteBasePath')) {
+    log(
+      'Setting server.basePath without server.rewriteBasePath is deprecated until Kibana 7.0, ' +
+      'when the default behavior of server.basePath will change and Kibana will start expecting ' +
+      'that all requests start with the server.basePath rather expecting you to rewrite the ' +
```

Comment:
nit, rather _than_ expecting

Reply:
Also maybe a nit, I found it hard to follow this sentence. Would it be easier to read if it said, 

> You should set server.basePath along with server.rewriteBasePath. From Kibana 7.0 onward, server.rewriteBasePath defaults to true, even if not set; Kibana will start expecting that all requests start with the server.basePath rather than expecting you to rewrite the requests in your reverse proxy. Explicitly set server.rewriteBasePath to false to preserve the current behavior and silence this warning.

Reply:
Thanks @archanid, I like your phasing in certain parts a lot more. I'll merge the two.